bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/Workbench.java

```diff
@@ -1483,6 +1489,63 @@ IWorkbenchWindow createWorkbenchWindow(IAdaptable input, IPerspectiveDescriptor
 		return result;
 	}
 
+	private void placeNearActiveShell(MWindow window) {
+		if (getDisplay() == null)
+			return;
+
+		Shell activeShell = getDisplay().getActiveShell();
+		if (activeShell == null)
+			return;
+
+		Rectangle newShellBounds = activeShell.getBounds();
+		Monitor currentMonitor = findMonitorThatContains(newShellBounds.x, newShellBounds.y);
+
+		Rectangle currentMonitorBounds = currentMonitor.getBounds();
+
+		final int offsetToExistingShell = 100;
+		newShellBounds.x += offsetToExistingShell;
+		newShellBounds.y += offsetToExistingShell;
+		moveIntoBounds(newShellBounds, currentMonitorBounds);
```

Comment:
When testing the patch, the behavior due to these lines seems to be unintended to me: It takes the size of the original shell instead of the new shell (initialized with a default size) for computing the bounds.

I'd propose to initialize the new shell bounds instead as follows to achieve the expected behavior:
```java
final int offsetToExistingShell = 100;
Rectangle newShellBounds = new Rectangle(
				activeShell.getBounds().x + offsetToExistingShell,
				activeShell.getBounds().y + offsetToExistingShell,
				window.getWidth(),
				window.getHeight());
```

Reply:
Thanks for hint. I took your suggestion in 3a3210d663b582050f6d192d4e27bfd728425cea